panther_lights/src/controller_node.py

```diff
@@ -189,15 +270,78 @@ def _set_image_animation_cb(
         return SetLEDImageAnimationResponse(True, f'Successfully set custom animation')
 
     def _update_animations_cb(self, req: TriggerRequest) -> TriggerResponse:
-        self._update_animations()
+        try:
+            self._update_animations()
+        except (KeyError, FileNotFoundError) as err:
+            return TriggerResponse(False, f'Failed to update animations: {err}')
         return TriggerResponse(True, 'Animations updated successfully')
 
     def _get_animation_by_id(self, animation_id: int) -> PantherAnimation:
+        try:
+            animation = deepcopy(self._animations[animation_id])
+            animation.reset_time()
+            return animation
+        except KeyError:
+            raise ValueError(f'No Animation with id: {animation_id}')
+
+    def _add_animation_to_queue(self, animation: PantherAnimation) -> None:
+        self._anim_queue.put(animation)
+        if animation.repeating:
+            self._anim_queue.remove(self._default_animation)
+            self._default_animation = animation
 
-        animation = PantherAnimation()
+    def _get_image_animation_description(self, animation: LEDImageAnimation) -> dict:
+        if not animation.image:
+            raise Exception('Missing required field \'image\'')
+
+        animation_description = {
+            'image': animation.image,
+            'duration': animation.duration,
+            'repeat': animation.repeat,
+            'brightness': animation.brightness,
+        }
+
+        if animation.color:
+            animation_description.update({'color': animation.color})
```

Comment:
```suggestion
            animation_description['color'] = animation.color
```
This is more normal way of specifying this I think